core/source_map.rs

```diff
@@ -153,13 +153,18 @@ impl SourceMapper {
           match resolve_url(source_file_name) {
             Ok(m) if m.scheme() == "blob" => None,
             Ok(m) => Some(m.to_string()),
-            Err(_) => {
-              // Try to resolve as a relative path from the module URL
-              resolve_url(file_name)
-                .ok()
-                .and_then(|base_url| base_url.join(source_file_name).ok())
-                .map(|resolved| resolved.to_string())
-            }
+            Err(_) => resolve_url(file_name)
+              .ok()
+              .and_then(|base_url| base_url.join(source_file_name).ok())
+              .and_then(|resolved| {
+                if resolved.scheme() == "file"
+                  && resolved.to_file_path().ok()?.exists()
```

Comment:
I don't think this will work - it hits disk and ignores the permission checks - maybe we need to have another method on the module loader that allows to check if the file actually exists?

Reply:
I feel like we should revert to this. Ignoring permission checks is fine.